k8s-plugin/src/main/kotlin/com/amazon/spinnaker/keel/k8s/resolver/KustomizeResourceHandler.kt

```diff
@@ -1,66 +1,85 @@
 import com.amazon.spinnaker.keel.k8s.model.K8sObjectManifest
+import com.amazon.spinnaker.keel.k8s.model.KustomizeResourceSpec
+import com.amazon.spinnaker.keel.k8s.resolver.FluxManifestUtil.generateGitRepoManifest
 import com.amazon.spinnaker.keel.k8s.service.CloudDriverK8sService
-import com.fasterxml.jackson.module.kotlin.jacksonObjectMapper
-import com.fasterxml.jackson.module.kotlin.readValue
 import com.netflix.spinnaker.keel.api.Resource
 import com.netflix.spinnaker.keel.api.actuation.TaskLauncher
 import com.netflix.spinnaker.keel.api.plugins.Resolver
 import com.netflix.spinnaker.keel.api.support.EventPublisher
 import com.netflix.spinnaker.keel.orca.OrcaService
+import com.netflix.spinnaker.keel.persistence.KeelRepository
+import com.netflix.spinnaker.keel.plugin.CannotResolveDesiredState
+import com.netflix.spinnaker.kork.exceptions.IntegrationException
 
 class KustomizeResourceHandler(
     override val cloudDriverK8sService: CloudDriverK8sService,
     override val taskLauncher: TaskLauncher,
     override val eventPublisher: EventPublisher,
     orcaService: OrcaService,
-    override val resolvers: List<Resolver<*>>
-) : GenericK8sResourceHandler<KustomizeResourceSpec, K8sObjectManifest>(
-    cloudDriverK8sService, taskLauncher, eventPublisher, orcaService, resolvers
+    override val resolvers: List<Resolver<*>>,
+    override val repository: KeelRepository
+) : BaseFluxHandler<KustomizeResourceSpec, K8sObjectManifest>(
+    cloudDriverK8sService, taskLauncher, eventPublisher, orcaService, resolvers, repository
 ) {
     override val supportedKind = KUSTOMIZE_RESOURCE_SPEC_V1
 
     public override suspend fun toResolvedType(resource: Resource<KustomizeResourceSpec>): K8sObjectManifest {
+        verifyKustomizeResource(resource)
+        log.debug("attempting to resolve resource for git")
 
-        if (resource.spec.template.apiVersion == FLUX_KUSTOMIZE_API_VERSION && resource.spec.template.kind == FLUX_KUSTOMIZE_KIND) {
-            return super.toResolvedType(resource)
+        val (artifact, deliveryConfig) = getArtifactAndConfig(resource)
+        // Kustomize controller only supports GitRepository as source
+        if (artifact !is GitRepoArtifact) {
+            throw InvalidArtifact("provided artifact is not a GitRepository artifact.")
         }
+        val spec = resource.spec.template.spec
+        val environment = repository.environmentFor(resource.id)
+        val version = repository.latestVersionApprovedIn(
+            deliveryConfig, artifact, environment.name
+        ) ?: throw NoVersionAvailable(artifact.name, artifact.type)
+        log.debug("found deployable version: $version")
+        val resolvedArtifact = resolveArtifactSpec(resource, artifact)
+        val sourceRef = mapOf(
+            "kind" to resolvedArtifact.kind,
+            "name" to "${resolvedArtifact.name}-${environment.name}",
+            "namespace" to resolvedArtifact.namespace
```

Comment:
some of these strings for `kind`, `name`, `namespace`, we have defined as constants. shall we use them instead of hardcoding? also let's define constants for `sourceRef` and `spec` and `chart` as well following Java best practices.